src/java.base/share/classes/java/math/BigDecimal.java

```diff
@@ -303,6 +303,20 @@
  * @since 1.1
  */
 public class BigDecimal extends Number implements Comparable<BigDecimal> {
+    /*
+     * Let l = log_2(10).
+     * Then, L < l < L + ulp(L) / 2, that is, L = roundTiesToEven(l).
```

Comment:
It doesn't matter in terms of the code, but shouldn't this be something like:

L - (ulp(L)) < l < L  ulp(L)

In other words, without further checking, it isn't clear that L is the lower-bound of the two double value bracketing l.

(If the ulp function being discussed were the real-valued version than L +/- ulp(l) would also be a reasonable formulation.)

Reply:
While it isn't obvious from the definition, high precision computations reveal that the inequalities `L < l < L + ulp(L) / 2` do indeed hold.
But as noted, this is not really relevant for the rest of the analysis.